drivers/sensor/sngcja5/sngcja5.c

```diff
@@ -0,0 +1,179 @@
+{
+	uint8_t buf[4];
+	int ret;
+
+	ret = read_bytes(dev, addr, &buf[0], 4);
+	if (ret) {
+		LOG_ERR("read_bytes() @ i2c failed");
+		return -EIO;
+	}
+
+	*value = (buf[3] << 24) | (buf[2] << 16) | (buf[1] << 8) | buf[0];
+
+	return 0;
+}
+
+static int read_register_2(const struct i2c_dt_spec *dev, uint8_t addr, uint16_t *value)
+{
+	uint8_t buf[2];
+	int ret;
+
+	ret = read_bytes(dev, addr, &buf[0], 2);
+	if (ret) {
+		LOG_ERR("read_bytes() @ i2c failed");
+		return -EIO;
+	}
+
+	*value = (buf[1] << 8) | buf[0];
+
+	return 0;
+}
+
+static int sngcja5_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct sngcja5_data *device_data = dev->data;
+	const struct sngcja5_config *device_config = dev->config;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	/* Read out all relevant sensor value registers */
+	if (read_register_4(&device_config->i2c, SNGCJA5_PM10_LL, &(device_data->pm1_0))) {
+		return -EIO;
+	}
+	if (read_register_4(&device_config->i2c, SNGCJA5_PM25_LL, &(device_data->pm2_5))) {
+		return -EIO;
+	}
+	if (read_register_4(&device_config->i2c, SNGCJA5_PM100_LL, &(device_data->pm10_0))) {
+		return -EIO;
+	}
+	if (read_register_2(&device_config->i2c, SNGCJA5_05_L, &(device_data->pc0_5))) {
+		return -EIO;
+	}
+	if (read_register_2(&device_config->i2c, SNGCJA5_10_L, &(device_data->pc1_0))) {
+		return -EIO;
+	}
+	if (read_register_2(&device_config->i2c, SNGCJA5_25_L, &(device_data->pc2_5))) {
+		return -EIO;
+	}
+	if (read_register_2(&device_config->i2c, SNGCJA5_50_L, &(device_data->pc5_0))) {
+		return -EIO;
+	}
+	if (read_register_2(&device_config->i2c, SNGCJA5_75_L, &(device_data->pc7_5))) {
+		return -EIO;
+	}
+	if (read_register_2(&device_config->i2c, SNGCJA5_100_L, &(device_data->pc10_0))) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int sngcja5_channel_get(const struct device *dev, enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct sngcja5_data *device_data = dev->data;
+	int32_t uval;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+	if (chan != SENSOR_CHAN_ALL) {
+		return -ENOTSUP;
+	}
+
+	val[0].val1 = device_data->pc0_5;
+	val[0].val2 = 0;
+	val[1].val1 = device_data->pc1_0;
+	val[1].val2 = 0;
+	val[2].val1 = device_data->pc2_5;
+	val[2].val2 = 0;
+	val[3].val1 = device_data->pc5_0;
+	val[3].val2 = 0;
+	val[4].val1 = device_data->pc7_5;
+	val[4].val2 = 0;
+	val[5].val1 = device_data->pc10_0;
+	val[5].val2 = 0;
+
+	uval = (int32_t)device_data->pm1_0;
+	val[6].val1 = uval / SNGCJA5_SCALE_FACTOR;
+	val[6].val2 = uval % SNGCJA5_SCALE_FACTOR;
+	uval = (int32_t)device_data->pm2_5;
+	val[7].val1 = uval / SNGCJA5_SCALE_FACTOR;
+	val[7].val2 = uval % SNGCJA5_SCALE_FACTOR;
+	uval = (int32_t)device_data->pm10_0;
+	val[8].val1 = uval / SNGCJA5_SCALE_FACTOR;
+	val[8].val2 = uval % SNGCJA5_SCALE_FACTOR;
+
+	return 0;
+}
+
+int sngcja5_init(const struct device *dev)
+{
+	const struct sngcja5_config *config = dev->config;
+
+	if (!device_is_ready(config->i2c.bus)) {
```

Comment:
`i2c_is_ready_dt`

Reply:
Pushed as requested in latest pull request update.